src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -63,25 +63,38 @@ function getFieldsForTypes(types: string[], searchFields?: string[]) {
  */
 function getClauseForType(
   registry: ISavedObjectTypeRegistry,
-  namespace: string | undefined,
+  namespaces: string[] = ['default'],
   type: string
 ) {
   if (registry.isMultiNamespace(type)) {
     return {
       bool: {
-        must: [{ term: { type } }, { term: { namespaces: namespace ?? 'default' } }],
+        must: [{ term: { type } }, { terms: { namespaces } }],
         must_not: [{ exists: { field: 'namespace' } }],
       },
     };
-  } else if (namespace && registry.isSingleNamespace(type)) {
+  } else if (registry.isSingleNamespace(type)) {
+    const should: Array<Record<string, any>> = [];
+    const eligibleNamespaces = namespaces.filter((namespace) => namespace !== 'default');
+    if (eligibleNamespaces.length > 0) {
+      should.push({ terms: { namespace: eligibleNamespaces } });
+    }
+    if (namespaces.includes('default')) {
+      should.push({ bool: { must_not: [{ exists: { field: 'namespace' } }] } });
+    }
+    if (should.length === 0) {
+      throw new Error('cannot specify empty namespaces array');
+    }
```

Comment:
Can we safely exclude this from happening for multi namespace types? I thought we should fail for `namespaces.length === 0` for both cases at the top of the function, maybe even if SavedObjectsRepository#find? The HTTP API and https://github.com/elastic/kibana/blob/36dd1b4fbc3cb775669ceb18ee967f663f5fd313/x-pack/plugins/spaces/server/saved_objects/spaces_saved_objects_client.ts#L159 prevents `namespaces=[]`

Reply:
I'll move this check to the top of the function, good call. I could duplicate this check within the repository, but it seems unnecessary at this point (IMO). Happy to revisit though. We have an explicit check within the `spaces_saved_objects_client` because we want to ensure the behavior is consistent from an authorization perspective, so as to not leak more information than necessary